ros2bag/ros2bag/verb/record.py

```diff
@@ -197,6 +197,11 @@ def main(self, *, args):  # noqa: D102
             return print_error('Invalid choice: Cannot specify compression format '
                                'without a compression mode.')
 
+        if args.compression_mode == 'message' and args.storage == 'mcap':
+            return print_error('Invalid choice: compression_mode \'message\' is only supported '
+                               'with the sqlite3 storage plugin. When using the MCAP storage plugin, you can '
+                               'enable chunk compression by setting `compression: \'Zstd\'` in storage config')
+
```

Comment:
I agree with @MichaelOrlov, why does generic CLI part need to be aware of the implementation details what is supported or not? instead, can we just call API and then catch the error or exception that response should tell why it fails?

@facontidavide if i may, what is the background for this PR? is there any specific issue to avoid this condition?

Reply:
The background of PR is a request from Foxglove.
I am helping them with the WHAT and HOW, without questioning too much the WHY :wink: 

If you want to remove the code in the `record.py`, let me know, and I will

Reply:
I do not think you need to remove this in necessary, thanks for sharing the information!